app/scripts/controllers/createClientBuild.js

```diff
@@ -0,0 +1,259 @@
+        }
+      };
+
+      if(clientConfig.clientCredentialsName) {
+        buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push({name: 'BUILD_CREDENTIAL_ID', value: $scope.projectName + '-' + clientConfig.clientCredentialsName});
+      }
+
+      if(clientConfig.credentialsAlias) {
+        buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push({name: 'BUILD_CREDENTIAL_ALIAS', value: clientConfig.credentialsAlias});
+      }
+
+      if (clientConfig.gitCredentialsName) {
+        buildConfig.spec.source.sourceSecret = {
+          name: clientConfig.gitCredentialsName
+        };
+      }
+
+      return buildConfig;
+    };
+
+    var createGitCredentialsSecret = function(clientConfig) {
+      var secret = {
+        apiVersion: APIService.toAPIVersion(secretsVersion),
+        kind: 'Secret',
+        metadata: {
+          name: clientConfig.gitCredentialsName,
+          labels:  {
+            'mobile-client-build': 'true'
+          }
+        },
+        type: clientConfig.authType,
+        stringData: {}
+      };
+
+      switch (clientConfig.authType) {
+        case 'kubernetes.io/basic-auth':
+          secret.stringData.password = clientConfig.passwordToken;
+          secret.stringData.username = clientConfig.username;
+          secret.type = 'Opaque';
+          break;
+        case 'kubernetes.io/ssh-auth':
+          secret.stringData['ssh-privatekey'] = clientConfig.privateKey;
+          break;
+      }
+
+      return secret;
+    };
+
+    var createClientCredentialsSecret = function(clientConfig) {
+      var secret = {
+        apiVersion: APIService.toAPIVersion(secretsVersion),
+        kind: 'Secret',
+        metadata: {
+          name:  clientConfig.clientCredentialsName,
+          labels:  {
+            'mobile-client-build': 'true',
+            'credential.sync.jenkins.openshift.io': 'true'
+          }
+        },
+        type: 'Opaque',
+        stringData: {}
+      };
+      var secretData = {
+        password: clientConfig.clientCredentialsPassword
+      };
+
+      if (clientConfig.buildPlatform === 'android') {
+        secretData['certificate'] = clientConfig.clientCredentials;
+      }
+
+      if (clientConfig.buildPlatform === 'ios') {
+        secretData['developer-profile'] = clientConfig.clientCredentials;
+      }
+      
+      // Base64 encode the values.
+      secret.data = _.mapValues(secretData, window.btoa);
+
+      return secret;
+    };
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        $scope.context = context;
+
+        DataService.get(APIService.getPreferredVersion('mobileclients'), $routeParams.mobileclient, context).then(function(mobileClient) {
+          $scope.mobileClient = mobileClient;
+          var clientType = _.get(mobileClient, 'spec.clientType').toLowerCase();
+          var buildPlatform = $scope.buildTypeMap[clientType].buildPlatforms[0];
```

Comment:
`.buildPlatforms[0]` - this just fills out initial values in the form and uses android as a default.
`clientType` comes from our mobile clients apbs, it's very unlikely that it will be invalid.